"ld: unknown option: -z" on OSX

I am getting this error when trying to compile ESBMC on OSX: `ld: unknown option: -z`. This happens when `cmake` tries to configure Z3:

`CMake Error at src/solvers/z3/CMakeLists.txt:34 (message):
  Could not use Z3:`

Also when I try compiling without Z3, this error appears eventually during the ESBMC compilation.

--- a/src/solvers/smtlib/CMakeLists.txt
+++ b/src/solvers/smtlib/CMakeLists.txt
@@ -9,7 +9,7 @@ configure_file(smtlib_tok.lpp  ${CMAKE_BINARY_DIR}/smtlib_tok.lpp COPYONLY)
 
 add_library(smtlib smtlib_conv.cpp ${CMAKE_CURRENT_BINARY_DIR}/smtlib.cpp ${CMAKE_CURRENT_BINARY_DIR}/lexer.cpp)
 target_link_libraries(smtlib fmt::fmt)
-target_include_directories(smtlib     
+target_include_directories(smtlib
     PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}
     PRIVATE ${CMAKE_CURRENT_BINARY_DIR}
     PRIVATE ${CMAKE_BINARY_DIR}/src
@@ -21,7 +21,11 @@ target_include_directories(smtlib
 # could be very large and the std::list destructors use a lot of stack space
 set(stacksize 20971520)  # 20 MB ought to be enough for everyone
 if(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
-  set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /STACK:${stacksize}" CACHE STRING "Linker flags" FORCE)
+set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /STACK:${stacksize}" CACHE STRING "Linker flags" FORCE)
+elseif(CMAKE_CXX_COMPILER_ID STREQUAL "Clang" OR CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang")
+  set(stacksize 0x1400000) # macOS uses hex by default (see Issue #1212)
+  set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,-stack_size,${stacksize}" CACHE STRING "Linker flags" FORCE)
 else()
+  # GNU probably
   set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,-z,stack-size=${stacksize}" CACHE STRING "Linker flags" FORCE)
 endif()

--- a/src/solvers/smtlib/CMakeLists.txt
+++ b/src/solvers/smtlib/CMakeLists.txt
@@ -21,7 +21,7 @@ target_include_directories(smtlib
 # could be very large and the std::list destructors use a lot of stack space
 set(stacksize 20971520)  # 20 MB ought to be enough for everyone
 if(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
-set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /STACK:${stacksize}" CACHE STRING "Linker flags" FORCE)
+  set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /STACK:${stacksize}" CACHE STRING "Linker flags" FORCE)
 elseif(CMAKE_CXX_COMPILER_ID STREQUAL "Clang" OR CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang")
   set(stacksize 0x1400000) # macOS uses hex by default (see Issue #1212)
   set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,-stack_size,${stacksize}" CACHE STRING "Linker flags" FORCE)

